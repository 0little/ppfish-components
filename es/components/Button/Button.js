var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
import * as React from 'react';
import { findDOMNode } from 'react-dom';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon/index';
import './style/index.less';
const rxTwoCNChar = /^[\u4e00-\u9fa5]{2}$/;
const isTwoCNChar = rxTwoCNChar.test.bind(rxTwoCNChar);
function isString(str) {
    return typeof str === 'string';
}
// Insert one space between two chinese characters automatically.
function insertSpace(child, needInserted) {
    // Check the child if is undefined or null.
    if (child == null) {
        return;
    }
    const SPACE = needInserted ? ' ' : '';
    // strictNullChecks oops.
    if (typeof child !== 'string' && typeof child !== 'number' &&
        isString(child.type) && isTwoCNChar(child.props.children)) {
        return React.cloneElement(child, {}, child.props.children.split('').join(SPACE));
    }
    if (typeof child === 'string') {
        if (isTwoCNChar(child)) {
            child = child.split('').join(SPACE);
        }
        return React.createElement("span", null, child);
    }
    return child;
}
export default class Button extends React.Component {
    constructor(props) {
        super(props);
        this.handleClick = e => {
            // Add click effect
            this.setState({ clicked: true });
            clearTimeout(this.timeout);
            this.timeout = window.setTimeout(() => this.setState({ clicked: false }), 500);
            const onClick = this.props.onClick;
            if (onClick) {
                onClick(e);
            }
        };
        this.state = {
            loading: props.loading,
            clicked: false,
            hasTwoCNChar: false,
        };
    }
    componentDidMount() {
        this.fixTwoCNChar();
    }
    componentWillReceiveProps(nextProps) {
        const currentLoading = this.props.loading;
        const loading = nextProps.loading;
        if (currentLoading) {
            clearTimeout(this.delayTimeout);
        }
        if (typeof loading !== 'boolean' && loading && loading.delay) {
            this.delayTimeout = window.setTimeout(() => this.setState({ loading }), loading.delay);
        }
        else {
            this.setState({ loading });
        }
    }
    componentDidUpdate() {
        this.fixTwoCNChar();
    }
    componentWillUnmount() {
        if (this.timeout) {
            clearTimeout(this.timeout);
        }
        if (this.delayTimeout) {
            clearTimeout(this.delayTimeout);
        }
    }
    fixTwoCNChar() {
        // Fix for HOC usage like <FormatMessage />
        const node = findDOMNode(this);
        const buttonText = node.textContent || node.innerText;
        if (this.isNeedInserted() && isTwoCNChar(buttonText)) {
            if (!this.state.hasTwoCNChar) {
                this.setState({
                    hasTwoCNChar: true,
                });
            }
        }
        else if (this.state.hasTwoCNChar) {
            this.setState({
                hasTwoCNChar: false,
            });
        }
    }
    isNeedInserted() {
        const { icon, children } = this.props;
        return React.Children.count(children) === 1 && !icon;
    }
    render() {
        const _a = this.props, { type, shape, size, className, children, icon, prefixCls, ghost, loading: _loadingProp } = _a, rest = __rest(_a, ["type", "shape", "size", "className", "children", "icon", "prefixCls", "ghost", "loading"]);
        const { loading, clicked, hasTwoCNChar } = this.state;
        // large => lg
        // small => sm
        let sizeCls = '';
        switch (size) {
            case 'large':
                sizeCls = 'lg';
                break;
            case 'small':
                sizeCls = 'sm';
            default:
                break;
        }
        const classes = classNames(prefixCls, className, {
            [`${prefixCls}-${type}`]: type,
            [`${prefixCls}-${shape}`]: shape,
            [`${prefixCls}-${sizeCls}`]: sizeCls,
            [`${prefixCls}-icon-only`]: !children && icon,
            [`${prefixCls}-loading`]: loading,
            [`${prefixCls}-clicked`]: clicked,
            [`${prefixCls}-background-ghost`]: ghost,
            [`${prefixCls}-two-chinese-chars`]: hasTwoCNChar,
        });
        const iconType = loading ? 'load-line' : icon;
        const iconNode = iconType ? React.createElement(Icon, { type: iconType, spinning: loading }) : null;
        const kids = (children || children === 0)
            ? React.Children.map(children, child => insertSpace(child, this.isNeedInserted())) : null;
        if ('href' in rest) {
            return (React.createElement("a", Object.assign({}, rest, { className: classes, onClick: this.handleClick }),
                iconNode,
                kids));
        }
        else {
            // React does not recognize the `htmlType` prop on a DOM element. Here we pick it out of `rest`.
            const { htmlType } = rest, otherProps = __rest(rest, ["htmlType"]);
            return (React.createElement("button", Object.assign({}, otherProps, { type: htmlType || 'button', className: classes, onClick: this.handleClick }),
                iconNode,
                kids));
        }
    }
}
Button.__FISHD_BUTTON = true;
Button.defaultProps = {
    prefixCls: 'fishd-btn',
    loading: false,
    ghost: false,
};
Button.propTypes = {
    type: PropTypes.string,
    shape: PropTypes.oneOf(['circle', 'circle-outline']),
    size: PropTypes.oneOf(['large', 'default', 'small']),
    htmlType: PropTypes.oneOf(['submit', 'button', 'reset']),
    onClick: PropTypes.func,
    loading: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
    className: PropTypes.string,
    icon: PropTypes.string,
};
