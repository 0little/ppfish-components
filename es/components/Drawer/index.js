function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import RcDrawer from './src';
import './style/index.less';

var Drawer =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Drawer, _React$Component);

  function Drawer(props) {
    var _this;

    _classCallCheck(this, Drawer);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Drawer).call(this, props));

    _defineProperty(_assertThisInitialized(_this), "handleMaskClick", function (e) {
      _this.props.onMaskClick(e);
    });

    _defineProperty(_assertThisInitialized(_this), "handleChange", function (status) {
      _this.props.onChange(status);
    });

    _defineProperty(_assertThisInitialized(_this), "handleHandleClick", function (e) {
      _this.props.onHandleClick(e);
    });

    _defineProperty(_assertThisInitialized(_this), "handleCloseClick", function (e) {
      _this.props.onCloseClick(e);
    });

    return _this;
  }

  _createClass(Drawer, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          className = _this$props.className,
          wrapperClassName = _this$props.wrapperClassName,
          width = _this$props.width,
          height = _this$props.height,
          visible = _this$props.visible,
          placement = _this$props.placement,
          getContainer = _this$props.getContainer,
          style = _this$props.style,
          mask = _this$props.mask,
          maskStyle = _this$props.maskStyle,
          handler = _this$props.handler,
          level = _this$props.level,
          ease = _this$props.ease,
          duration = _this$props.duration,
          closed = _this$props.closed;
      return React.createElement(RcDrawer, {
        className: className,
        wrapperClassName: wrapperClassName,
        width: width,
        height: height,
        open: visible,
        closed: closed,
        placement: placement,
        getContainer: getContainer,
        showMask: mask,
        level: level,
        ease: ease,
        duration: duration,
        maskStyle: maskStyle,
        style: style,
        handler: handler,
        onMaskClick: this.handleMaskClick,
        onHandleClick: this.handleHandleClick,
        onChange: this.handleChange,
        onCloseClick: this.handleCloseClick
      }, this.props.children);
    }
  }]);

  return Drawer;
}(React.Component);

_defineProperty(Drawer, "propTypes", {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  wrapperClassName: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  visible: PropTypes.bool,
  closed: PropTypes.bool,
  placement: PropTypes.oneOf(['left', 'right', 'bottom', 'top']),
  getContainer: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.func, PropTypes.bool]),
  style: PropTypes.object,
  mask: PropTypes.bool,
  maskStyle: PropTypes.object,
  children: PropTypes.node,
  handler: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),
  level: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  ease: PropTypes.string,
  duration: PropTypes.string,
  onChange: PropTypes.func,
  onMaskClick: PropTypes.func,
  onHandleClick: PropTypes.func,
  onCloseClick: PropTypes.func
});

_defineProperty(Drawer, "defaultProps", {
  prefixCls: 'fishd-drawer',
  placement: 'right',
  onChange: function onChange() {},
  onMaskClick: function onMaskClick() {},
  onHandleClick: function onHandleClick() {}
});

export { Drawer as default };