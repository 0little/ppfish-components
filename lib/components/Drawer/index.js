"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _src = _interopRequireDefault(require("./src"));

require("./style/index.less");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Drawer =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Drawer, _React$Component);

  function Drawer(props) {
    var _this;

    _classCallCheck(this, Drawer);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Drawer).call(this, props));

    _defineProperty(_assertThisInitialized(_this), "handleMaskClick", function (e) {
      _this.props.onMaskClick(e);
    });

    _defineProperty(_assertThisInitialized(_this), "handleChange", function (status) {
      _this.props.onChange(status);
    });

    _defineProperty(_assertThisInitialized(_this), "handleHandleClick", function (e) {
      _this.props.onHandleClick(e);
    });

    _defineProperty(_assertThisInitialized(_this), "handleCloseClick", function (e) {
      _this.props.onCloseClick(e);
    });

    return _this;
  }

  _createClass(Drawer, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          prefixCls = _this$props.prefixCls,
          className = _this$props.className,
          wrapperClassName = _this$props.wrapperClassName,
          width = _this$props.width,
          height = _this$props.height,
          visible = _this$props.visible,
          placement = _this$props.placement,
          getContainer = _this$props.getContainer,
          style = _this$props.style,
          mask = _this$props.mask,
          maskStyle = _this$props.maskStyle,
          handler = _this$props.handler,
          level = _this$props.level,
          ease = _this$props.ease,
          duration = _this$props.duration,
          closed = _this$props.closed;
      return React.createElement(_src["default"], {
        className: className,
        wrapperClassName: wrapperClassName,
        width: width,
        height: height,
        open: visible,
        closed: closed,
        placement: placement,
        getContainer: getContainer,
        showMask: mask,
        level: level,
        ease: ease,
        duration: duration,
        maskStyle: maskStyle,
        style: style,
        handler: handler,
        onMaskClick: this.handleMaskClick,
        onHandleClick: this.handleHandleClick,
        onChange: this.handleChange,
        onCloseClick: this.handleCloseClick
      }, this.props.children);
    }
  }]);

  return Drawer;
}(React.Component);

exports["default"] = Drawer;

_defineProperty(Drawer, "propTypes", {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  wrapperClassName: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  visible: PropTypes.bool,
  closed: PropTypes.bool,
  placement: PropTypes.oneOf(['left', 'right', 'bottom', 'top']),
  getContainer: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.func, PropTypes.bool]),
  style: PropTypes.object,
  mask: PropTypes.bool,
  maskStyle: PropTypes.object,
  children: PropTypes.node,
  handler: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),
  level: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  ease: PropTypes.string,
  duration: PropTypes.string,
  onChange: PropTypes.func,
  onMaskClick: PropTypes.func,
  onHandleClick: PropTypes.func,
  onCloseClick: PropTypes.func
});

_defineProperty(Drawer, "defaultProps", {
  prefixCls: 'fishd-drawer',
  placement: 'right',
  onChange: function onChange() {},
  onMaskClick: function onMaskClick() {},
  onHandleClick: function onHandleClick() {}
});