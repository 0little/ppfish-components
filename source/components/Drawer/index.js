import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import RcDrawer from './src';

import './style/index.less';

export default class Drawer extends React.Component {
  static propTypes = {
    prefixCls: PropTypes.string,
    className: PropTypes.string,
    wrapperClassName: PropTypes.string,
    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    visible: PropTypes.bool,
    closed: PropTypes.bool,
    placement: PropTypes.oneOf(['left', 'right', 'bottom', 'top']),
    getContainer: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
      PropTypes.func,
      PropTypes.bool,
    ]),
    style: PropTypes.object,
    mask: PropTypes.bool,
    maskStyle: PropTypes.object,
    children: PropTypes.node,
    handler: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),
    level: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    onChange: PropTypes.func,
    onMaskClick: PropTypes.func,
    onHandleClick: PropTypes.func,
    onCloseClick: PropTypes.func
  };

  static defaultProps = {
    prefixCls: 'fishd-drawer',
    placement: 'right',
    onChange: () => { },
    onMaskClick: () => { },
    onHandleClick: () => { },
  };

  constructor(props) {
    super(props);
  }

  handleMaskClick = (e) => {
    this.props.onMaskClick(e);
  }

  handleChange = (status) => {
    this.props.onChange(status);
  }

  handleHandleClick = (e) => {
    this.props.onHandleClick(e);
  }

  handleCloseClick = (e) => {
    this.props.onCloseClick(e);
  }

  render() {
    const {
      prefixCls,
      className,
      wrapperClassName,
      width,
      height,
      visible,
      placement,
      getContainer,
      style,
      mask,
      maskStyle,
      handler,
      level,
      closed
    } = this.props;

    return (
      <RcDrawer
        className={className}
        wrapperClassName={wrapperClassName}
        width={width}
        height={height}
        open={visible}
        closed={closed}
        placement={placement}
        getContainer={getContainer}
        showMask={mask}
        level={level}
        maskStyle={maskStyle}
        style={style}
        handler={handler}
        onMaskClick={this.handleMaskClick}
        onHandleClick={this.handleHandleClick}
        onChange={this.handleChange}
        onCloseClick={this.handleCloseClick}
      >
        {this.props.children}
      </RcDrawer>
    );
  }
}
