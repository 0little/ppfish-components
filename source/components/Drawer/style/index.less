@import (reference) "../../../assets/css/index";
@import "./drawer";
