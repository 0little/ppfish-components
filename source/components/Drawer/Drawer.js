import React, {PureComponent} from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon/index.tsx';
import './style/Drawer.less';

/**
 * 抽屉组件
 * @prop {string} className   设置类名
 * @prop {object} style       设置样式
 * @prop {bool} visible       图表参数
 * @prop {object} onClose     图表事件
 * @prop {node} container     容器
 */
class Drawer extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    container: PropTypes.instanceOf(Element),
    onClose: PropTypes.func,
    style: PropTypes.object,
    visible: PropTypes.bool.isRequired,
    prefixCls: PropTypes.string
  };

  static defaultProps = {
    visible: false,
    prefixCls: 'fishd'
  };

  handleClose = e => {
    const {onClose} = this.props;
    onClose && onClose();
  };

  render() {
    const {children, className, style, visible, container, prefixCls} = this.props;
    const drawerClass = classNames(`${prefixCls}-drawer`, className, {
      "z-visible": visible
    });
    const el = (
      <div className={drawerClass} style={style}>
        <span className={`${prefixCls}-drawer-close`} onClick={this.handleClose}>
          <Icon className="iconfont" type="close-modal-line"/>
        </span>
        <div className={`${prefixCls}-drawer-container`}>{children}</div>
      </div>
    );
    return container ? ReactDOM.createPortal(el, container) : el;
  }
}

export default Drawer;
