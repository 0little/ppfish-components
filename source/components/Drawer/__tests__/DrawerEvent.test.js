import React from 'react';
import { mount } from 'enzyme';
import Drawer from '../index.tsx';
import Button from '../../Button/index.tsx';

class DrawerEventTester extends React.Component {
  constructor(props) {
    super(props);
    this.state = { visible: false };
  }

  componentDidMount() {
    this.setState({ visible: true }); // eslint-disable-line react/no-did-mount-set-state
  }

  onClose = () => {
    this.setState({
      visible: false,
    });
  };

  open = () => {
    this.setState({
      visible: true,
    });
  }

  render() {
    const { visible } = this.state;
    return (
      <div>
        <Button onClick={this.open}>open</Button>
        <Drawer
          visible={visible}
          onClose={this.onClose}
          getContainer={false}
          {...this.props}
        >
          Here is content of Drawer
        </Drawer>
      </div>
    );
  }
}


describe('Drawer', () => {
  it('render correctly', () => {
    const wrapper = mount(<DrawerEventTester />);
    const body = wrapper.find('.fishd-drawer-body').exists();

    expect(body).toBe(true);
    wrapper.find('button.fishd-btn').simulate('click');

    const content = wrapper.find('.fishd-drawer-body').getDOMNode().innerHTML;
    expect(content).toBe('Here is content of Drawer');

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('mask trigger onClose', () => {
    const wrapper = mount(<DrawerEventTester />);

    wrapper.find('button.fishd-btn').simulate('click');
    expect(wrapper.instance().state.visible).toBe(true);

    wrapper.find('.fishd-drawer-mask').simulate('click');
    expect(wrapper.instance().state.visible).toBe(false);
  });

  it('close button trigger onClose', () => {
    const wrapper = mount(<DrawerEventTester />);

    wrapper.find('button.fishd-btn').simulate('click');
    expect(wrapper.instance().state.visible).toBe(true);

    wrapper.find('.fishd-drawer-close').simulate('click');
    expect(wrapper.instance().state.visible).toBe(false);
  });

  it('maskClosable no trigger onClose', () => {
    const wrapper = mount(<DrawerEventTester maskClosable={false} />);

    wrapper.find('button.fishd-btn').simulate('click');
    expect(wrapper.instance().state.visible).toBe(true);

    wrapper.find('.fishd-drawer-mask').simulate('click');
    expect(wrapper.instance().state.visible).toBe(true);
  });

  it('destroyOnClose is true onClose', () => {
    const wrapper = mount(<DrawerEventTester destroyOnClose />);
    wrapper.find('button.fishd-btn').simulate('click');
    expect(wrapper.find('.fishd-drawer-wrapper-body').exists()).toBe(true);

    wrapper.setState({
      visible: false,
    });
    wrapper.find('.fishd-drawer-wrapper-body').simulate('transitionend');
    expect(wrapper.find('.fishd-drawer-wrapper-body').exists()).toBe(false);
  });

  it('no mask and no closable', () => {
    const wrapper = mount(<DrawerEventTester destroyOnClose />);

    wrapper.find('button.fishd-btn').simulate('click');
    expect(wrapper.instance().state.visible).toBe(true);

    wrapper.find('.fishd-drawer-close').simulate('click');
    expect(wrapper.instance().state.visible).toBe(false);
  });
});
