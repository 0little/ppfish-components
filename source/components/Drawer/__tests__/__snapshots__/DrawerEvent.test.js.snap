// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer render correctly 1`] = `
<div>
  <button
    class="fishd-btn fishd-btn-clicked"
    type="button"
  >
    <span>
      open
    </span>
  </button>
  <div
    class=""
  >
    <div
      class="fishd-drawer fishd-drawer-right fishd-drawer-open"
    >
      <div
        class="fishd-drawer-mask"
      />
      <div
        class="fishd-drawer-content-wrapper"
        style="width: 256px;"
      >
        <div
          class="fishd-drawer-content"
        >
          <div
            class="fishd-drawer-wrapper-body"
            style="overflow: auto; height: 100%;"
          >
            <button
              aria-label="Close"
              class="fishd-drawer-close"
            >
              <span
                class="fishd-drawer-close-x"
              />
            </button>
            <div
              class="fishd-drawer-body"
            >
              Here is content of Drawer
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
